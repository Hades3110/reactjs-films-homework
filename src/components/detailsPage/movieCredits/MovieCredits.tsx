import { useEffect, useState } from "react"
import { CreditInterface, getMovieCredits } from "../../../services/filmService"
import MovieCreditsItem from "./movieCreditItem/MovieCreditItem"
import styles from './movieCredits.module.scss'
import cx from 'classnames'


const MoviCredits = ({ id }: { id: number }) => {
    const [credits, setCredits] = useState<CreditInterface[]>([])

    useEffect(() => {
        id ? getMovieCredits(id).then(res => setCredits(res.cast)) : ''
    }, [id])
    console.log(credits)

    return (
        <div className={credits.length > 0 ? cx(styles.creditList, styles.scroll) : styles.creditList}>
            {credits.length > 0 ?
                credits.map((el) => {
                    return <MovieCreditsItem character={el.character} gender={el.gender} key={el.id} name={el.name} profile_path={el.profile_path} />
                })
                : 'No information'}
        </div>)
}

export default MoviCredits