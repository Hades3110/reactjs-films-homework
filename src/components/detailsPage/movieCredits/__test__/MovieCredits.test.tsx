import { create } from "react-test-renderer";
import MovieCredits from '../MovieCredits';

describe('MovieCredits Component', () => {

  it('MovieCredits snapshot test', () => {
    const component = create(<MovieCredits id={0} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});