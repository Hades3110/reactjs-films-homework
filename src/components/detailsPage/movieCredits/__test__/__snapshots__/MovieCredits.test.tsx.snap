// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCredits Component MovieCredits snapshot test 1`] = `
<div
  className="creditList"
>
  No information
</div>
`;
